base/genericmemory.jl

```diff
@@ -84,17 +84,16 @@ function _unsetindex!(A::MemoryRef{T}) where T
     MemT = typeof(mem)
     arrayelem = datatype_arrayelem(MemT)
     elsz = datatype_layoutsize(MemT)
-    isboxed = 1; isunion = 2
+    isbits = 0; isboxed = 1; isunion = 2
+    arrayelem == isbits && datatype_pointerfree(T::DataType) && return A
```

Comment:
Theoretically we could also make this more precise too (remove the XXX comment), and eliminate many of the other checks because we have reflection of the offset of firstptr among the various `datatype_npointer` reflection queries in runtime_internals.jl
```
julia> unsafe_load(Ptr{Base.DataTypeLayout}(Pair{Int,Any}.layout)).firstptr
1
```

Reply:
I'm going to let you do this in a separate PR.